src/table/use-sticky-columns.tsx

```diff
@@ -0,0 +1,230 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import React, { useLayoutEffect, useState, createRef, useEffect, useCallback, useMemo } from 'react';
+import { TableProps } from './interfaces';
+import clsx from 'clsx';
+interface CellOffsets {
+  first: number[];
+  last: number[];
+}
+interface StickyStyles {
+  sticky: {
+    left?: string;
+    right?: string;
+  };
+  stuck: {
+    paddingLeft?: string;
+  };
+}
+
+interface StickyColumnParams {
+  containerWidth: number | null;
+  hasSelection: boolean;
+  isWrapperScrollable: boolean;
+  stickyColumns?: TableProps.StickyColumns;
+  visibleColumnsLength: number;
+  tableRefObject: React.RefObject<HTMLTableElement>;
+  wrapperRefObject: React.RefObject<HTMLDivElement>;
+}
+
+export interface GetStickyColumnProperties {
+  isSticky: boolean;
+  isLastStickyLeft: boolean;
+  isLastStickyRight: boolean;
+  stickyStyles: StickyStyles;
+}
+
+// We allow the table to have a minimum of 148px of available space besides the sum of the widths of the sticky columns
+// This value is an UX recommendation and is approximately 1/3 of our smallest breakpoint (465px)
+const MINIMUM_SCROLLABLE_SPACE = 148;
+
+const useCellOffsets = (tableCellRefs: Array<React.RefObject<HTMLTableCellElement>>) => {
+  // This hook calculates the cumulative offsets of the cells in each column of the table, based on the widths of their siblings
+
+  const [cellOffsets, setCellOffsets] = useState<CellOffsets>({ first: [], last: [] });
```

Comment:
Would it make sense to calculate offsets not for all columns but only for stickyColumns.first and stickyColumns.last? In case the calculation is triggered often it might result in a performance improvement for table's with many columns.

Reply:
This makes total sense. It would require some slight refactoring, I'll add this as post-release action item.